internal/services/network/network_manager_verifier_workspace_reachability_analysis_intent_resource.go

```diff
@@ -0,0 +1,298 @@
+}
+
+func (ManagerVerifierWorkspaceReachabilityAnalysisIntentResource) ResourceType() string {
+	return "azurerm_network_manager_verifier_workspace_reachability_analysis_intent"
+}
+
+func (ManagerVerifierWorkspaceReachabilityAnalysisIntentResource) ModelObject() interface{} {
+	return &ManagerVerifierWorkspaceReachabilityAnalysisIntentResourceModel{}
+}
+
+type ManagerVerifierWorkspaceReachabilityAnalysisIntentResourceModel struct {
+	Description           string                                                        `tfschema:"description"`
+	DestinationResourceId string                                                        `tfschema:"destination_resource_id"`
+	IpTraffic             []ManagerVerifierWorkspaceReachabilityAnalysisIntentIpTraffic `tfschema:"ip_traffic"`
+	Name                  string                                                        `tfschema:"name"`
+	SourceResourceId      string                                                        `tfschema:"source_resource_id"`
+	VerifierWorkspaceId   string                                                        `tfschema:"verifier_workspace_id"`
+}
+
+type ManagerVerifierWorkspaceReachabilityAnalysisIntentIpTraffic struct {
+	DesticationIps   []string                                      `tfschema:"destination_ips"`
+	DesticationPorts []string                                      `tfschema:"destination_ports"`
+	Protocols        []reachabilityanalysisintents.NetworkProtocol `tfschema:"protocols"`
+	SourceIps        []string                                      `tfschema:"source_ips"`
+	SourcePorts      []string                                      `tfschema:"source_ports"`
+}
+
+func (ManagerVerifierWorkspaceReachabilityAnalysisIntentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9\_\.\-]{1,64}$`),
+				"`name` must be between 1 and 64 characters long and can only contain letters, numbers, underscores(_), periods(.), and hyphens(-).",
+			),
+		},
+
+		"verifier_workspace_id": commonschema.ResourceIDReferenceRequiredForceNew(&reachabilityanalysisintents.VerifierWorkspaceId{}),
+
+		"source_resource_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.Any(
+				commonids.ValidateSubnetID,
+				commonids.ValidatePublicIPAddressID,
+				commonids.ValidateVirtualMachineID,
+			),
+		},
+
+		"destination_resource_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.Any(
+				commonids.ValidateSubnetID,
+				commonids.ValidatePublicIPAddressID,
+				commonids.ValidateVirtualMachineID,
+				commonids.ValidateSqlServerID,
+				commonids.ValidateStorageAccountID,
+				cosmosdb.ValidateDatabaseAccountID,
+			),
+		},
+
+		"ip_traffic": {
```

Comment:
how many of these are allowed? looking at the docs from https://learn.microsoft.com/en-us/azure/virtual-network-manager/how-to-verify-reachability-with-virtual-network-verifier#create-a-reachability-analysis-intent
it seems like "an intent" only has one?